pkg/oc/admin/router/router.go

```diff
@@ -684,11 +684,15 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out, errout io.Write
 		env["ROUTER_CANONICAL_HOSTNAME"] = cfg.RouterCanonicalHostname
 	}
 	// automatically start the internal metrics agent if we are handling a known type
-	if cfg.Type == "haproxy-router" && cfg.StatsPort != 0 {
-		env["ROUTER_LISTEN_ADDR"] = fmt.Sprintf("0.0.0.0:%d", cfg.StatsPort)
-		env["ROUTER_METRICS_TYPE"] = "haproxy"
-		env["ROUTER_METRICS_TLS_CERT_FILE"] = "/etc/pki/tls/metrics/tls.crt"
-		env["ROUTER_METRICS_TLS_KEY_FILE"] = "/etc/pki/tls/metrics/tls.key"
+	if cfg.Type == "haproxy-router" {
+		env["ROUTER_CONFIG_MANAGER"] = "haproxy-manager"
```

Comment:
If everyone is going to get these just make these the new default on the command.  Don't force people to change their config and make a pointless choice.  Otherwise you'll have to add this to the upgrade scripts.

Reply:
ok will do.

Reply:
hmm, so because the template router is used for f5 (going away but still) as well as nginx, this needs to be either set here since its haproxy specific or we need to pass down the type of router to the infra command (that's not done today). I can either pass down this new setting or then the type and set the defaults based on that in the infra command ... but in either case it still means upgrading existing haproxy-router deployments. So which do you prefer?

Reply:
I think if the config manager works, there’s no reason we wouldn’t have it on by default for haproxy.  However, we will likely default it off at first until we’re comfortable that we’ve caught all the edge cases and then default it on.  So ideally we just have a Boolean that has an internal default and if the user specifies it we obey their rule.  That avoids end users ending up with the flag in their config.